### PR TITLE
Doc-3374 public images

### PR DESCRIPTION
**Pull request by aimurphy**
New topic about convenience images

Preview: https://64c007b03a8e070274c06d3e--harness-developer.netlify.app/docs/continuous-integration/use-ci/set-up-build-infrastructure/public-docker-images

### CODE REVIEW

**Comment by codewdhruv** [comment]
Hi @aimurphy The naming convention for the image is modified. Please review the updated version [here](https://docs.google.com/document/d/1iIrju_tKlIdYzysxktcv4FVnUd5c4HeVMpQsFXC4lRA/edit?usp=sharing).

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/public-docker-images.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+---
+
+The [Harness Community](https://github.com/harness-community/) maintains pre-built, public Docker images that you can use to quickly set up a build environment. These images are heavily extended versions of official Docker images, and they include useful tools for CI pipelines.
+
+The benefits of using pre-built images include:
+
+* **Builds start faster:** Pre-built images are usually squashed, meaning they have have fewer and smaller layers. These images take less time to download, if they are not already cached on the host. By using pre-built images, your builds start faster.
+* **Bundled with useful tools:** Pre-built images include the minimum utilities required to run most builds on Harness CI along with language- or service-specific tools, such as Bundler for Ruby.
+
+:::info Difference from Harness CI images
```
Comment (StrawberryPad) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/public-docker-images.md:14]: This is rendered in all caps for some reason, which is less readable. I've seen you use this 'note title' style before (which is really useful) where it wasn't in all caps (I think).

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/public-docker-images.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+Available images include a base image and several language and service images. The language and service images are derived from the base image.
+
+The Docker images are available on the [Harness Community Docker Hub](https://hub.docker.com/u/harnesscommunity).
+
+The source code for the images is available in the [Harness Community ci-images GitHub repository](https://github.com/harness-community/ci-images).
+
+### Common tools
+
+The following tools are installed on all public images:
+
```
Comment (StrawberryPad) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/public-docker-images.md:31]: Might be better to put the list below in a headerless multi-column table to eliminate the v long list.

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/public-docker-images.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+To use a language image in a pipeline, reference it in a step's `spec.image`, for example:
+
+```yaml
+                 spec:
+                   connectorRef: account.harnessImage ## Use the built-in connector or your own Docker connector.
+                   image: harnesscommunity/ruby:latest ## Reference the Docker Hub repo, image, and tag.
+                   shell: Sh
+                   command: bundle exec rake test ## This is an example command.
+```
+
```
Comment (StrawberryPad) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/public-docker-images.md:114]: Can the above be less indented?

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/public-docker-images.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+## Image updates
+
+The Harness Community updates the pre-built images as needed for bug fixes and enhancements. Updates can impact image functionality in your pipelines. Follow the `#continuous-integration` channel on [Harness Community Slack](https://developer.harness.io/community) to get notified about image updates.
+
+## Best practice: Use a specific image tag
+
+Images are based on the most-recent Ubuntu LTS Docker image, and they have the base libraries for the specified language for service. Using the most-specific image possible makes your builds more predictable by avoiding incorporating unwanted, upstream changes until you can test them.
```
Comment (StrawberryPad) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/public-docker-images.md:139]:
```suggestion
Images are based on the most-recent Ubuntu LTS Docker image, and they have the base libraries for the specified language or service. Using the most-specific image possible makes your builds more predictable by avoiding incorporating unwanted, upstream changes until you can test them.
```

Comment (StrawberryPad) [reply docs/continuous-integration/use-ci/set-up-build-infrastructure/public-docker-images.md:139]: Not sure if "most-recent" and "most-specific" is a thing (with dashes)... maybe in this specific tech domain?

**Review by StrawberryPad — APPROVED** [review]

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/public-docker-images.md (hunk 3 above)
Comment (aimurphy) [reply docs/continuous-integration/use-ci/set-up-build-infrastructure/public-docker-images.md:114]: it is indented to the correct indentation level if they copy/paste into their pipeline yaml.

**Pull request merged**